ms2deepscore/models/load_model.py

```diff
@@ -0,0 +1,54 @@
+from pathlib import Path
+from typing import Tuple, Union
+import h5py
+import json
+from tensorflow.python.keras.saving import hdf5_format
+
+from ms2deepscore.SpectrumBinner import SpectrumBinner
+from .SiameseModel import SiameseModel
+
+
+def load_model(filename: Union[str, Path]) -> SiameseModel:
+    """
+    Load a MS2DeepScore model (SiameseModel) from file.
+
+    For example:
+
+    .. code-block:: python
+
+        from ms2deepscore.models import load_model
+        model = load_model("model_file_xyz.hdf5")
+
+    Parameters
+    ----------
+    filename
+        Filename. Expecting saved SiameseModel.
+
+    """
+    with h5py.File(filename, mode='r') as f:
+        binner_json = f.attrs['spectrum_binner']
+        keras_model = hdf5_format.load_model_from_hdf5(f)
+
+    # Reconstitute spectrum_binner
+    binner_dict = json.loads(binner_json)
+    spectrum_binner = SpectrumBinner(binner_dict["number_of_bins"],
+                                     binner_dict["mz_max"], binner_dict["mz_min"],
+                                     binner_dict["peak_scaling"],
+                                     binner_dict["allowed_missing_percentage"])
+    spectrum_binner.peak_to_position = {int(key): value for key, value in binner_dict["peak_to_position"].items()}
+    spectrum_binner.known_bins = binner_dict["known_bins"]
+
+    # Extract parameters for SiameseModel
+    embedding_dim = keras_model.layers[2].output_shape[1]
+    base_dims = []
+    for layer in keras_model.layers[2].layers:
+        if "dense" in layer.name:
+            base_dims.append(layer.output_shape[1])
+        elif "dropout" in layer.name:
+            dropout_rate = layer.rate
```

Comment:
Would it make sense to move this to `SiameseModel`? Maybe add a class method: `from_keras_model()`?

Reply:
This is a good point and similar to the one about importing a SpectrumBinner from json. In both cases I feel that we have to adapt the __init__() method first. I'll look into it a bit.

Reply:
I now indeed moved it to `SiameseModel` in a way that makes this piece of code obsolete. 
Maybe you can have a quick look at it again @svenvanderburg?

Reply:
That looks good! :)

Reply:
Great, thanks!